### DISCUSSION TITLE
[v5] Problem with an extra column on a Model

### DISCUSSION THREAD

**Discussion by betaWeb** [adonisjs/core#1852] [Help] [answered]
Hi :)

I've a problem with the creation of an extra property on my model. 
I create my extra column as follow :
```typescript
@column()
public get total_net_of_taxes(): number {
  return this.total - this.charges
}
```

But when I call my query as below, the column doesn't shows up.
```typescript
Account
  .query()
  .whereRaw(`YEAR(created_at) = ${year}`)
```

Why ? I tried with `Account.all()` but I have the same result.

**Comment by crbast** [comment]
Did you create new migration file for the new column? or have you modified the migration related to the model?

**Reply by betaWeb** [reply]
> Did you create new migration file for the new column? or have you modified the migration related to the model?

Thanks you for your reply. 

Neither, I want to use extra columns (sort of computed properties) on my model as the documentation explains : https://preview.adonisjs.com/guides/models/columns#defining-columns

Perhaps I misenderstand something?

**Reply by crbast** [reply]
Sorry I misread 😖
Did you try [computed properties](https://preview.adonisjs.com/guides/models/serializing-models#computed-properties)?

Use `@column()` when you define table column

**Comment by hariprasath-yadav** [comment, marked as answer]
Your mistyped this
```
@column()
public get total_net_of_taxes(): number {
  return this.total - this.charges
}
```
try this
```
@computed()
public get total_net_of_taxes(): number {
  return this.total - this.charges
}
```
and also my one more suggestion, try to use the camel case function name
like **totalNetOfTaxes**
